### PR TITLE
Bc/Add find in PDF test

### PR DESCRIPTION
**Pull request by ben-c-at-moz**
# Description

Add test for Find Toolbar in PDFs, and some helpers to do more Find and PDF stuff

## Links

**TestRail:** https://testrail.stage.mozaws.net/index.php?/cases/view/127271
**Bugzilla:** https://bugzilla.mozilla.org/show_bug.cgi?id=1902754

## Type of change

Please delete options that are not relevant.

- [x] New Test
- [x] New POM
- [x] QoL: get_parent_of a given element, reliable fillable pdf fixture

# How does this resolve / make progress on that bug?

completes test

# Comments / Concerns

Nothing comes to mind

### CODE REVIEW

File: tests/find_toolbar/test_find_in_pdf.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+from selenium.webdriver import Firefox
+from selenium.webdriver.common.keys import Keys
+
+from modules.browser_object import FindToolbar
+from modules.page_base import BasePage
+from modules.page_object import GenericPdf
+
+WORD_SEGMENT = "authori"
+MATCH_TWO_COTEXT = "representative must complete"
```
Comment (ben-c-at-moz) [inline tests/find_toolbar/test_find_in_pdf.py:12]: Nb: I know this looks like "context" but I did actually mean "cotext": "the words in the environment immediately surrounding a word or phrase"

File: modules/page_base.py [hunk 2]
```diff
@@ -306,6 +306,14 @@ def get_elements(self, name: str, labels=[]):
         """
         return self.get_element(name, multiple=True, labels=labels)
 
+    def get_parent_of(self, name: str, labels=[]) -> WebElement:
```
Comment (sarinali) [inline modules/page_base.py:309]: very needed helper!

File: modules/page_object_generic_pdf.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+from modules.page_base import BasePage
+
+
+class GenericPdf(BasePage):
+    """
+    Generic POM for any page with an open PDF in it.
+    """
+
+    URL_TEMPLATE = "{pdf_url}"
```
Comment (sarinali) [inline modules/page_object_generic_pdf.py:9]: is this matched during runtime to a regex or just a placeholder? just wondering

File: tests/find_toolbar/test_find_in_pdf.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+
+class element_parent_has_text:
```
Comment (sarinali) [inline tests/find_toolbar/test_find_in_pdf.py:15]: would appreciate a 1 line comment explaining the purpose of this!

**Review by sarinali — APPROVED** [review]
lgtm, minor questions

File: modules/page_object_generic_pdf.py (hunk 3 above)
Comment (ben-c-at-moz) [reply modules/page_object_generic_pdf.py:9]: The `{pdf_url}` is replaced with the url passed to the POM, much like how we choose our category in `AboutPrefs`. Just lets you use it to open any PDF, and the POM should only contain things that are general PDF markup elements.

File: tests/find_toolbar/test_find_in_pdf.py (hunk 4 above)
Comment (ben-c-at-moz) [reply tests/find_toolbar/test_find_in_pdf.py:15]: On it!

**Review by sarinali — APPROVED** [review]
lgtm! everything passed

**Pull request merged**